cdap-app-fabric/src/main/java/co/cask/cdap/templates/AdapterPlugin.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates;
+
+import co.cask.cdap.api.templates.plugins.PluginClass;
+import co.cask.cdap.api.templates.plugins.PluginInfo;
+import co.cask.cdap.api.templates.plugins.PluginProperties;
+
+/**
+ * A container class for holding plugin information for an adapter instance.
+ */
+public final class AdapterPlugin {
+  private final PluginInfo pluginInfo;
+  private final PluginClass pluginClass;
+  private final PluginProperties properties;
+
+  AdapterPlugin(PluginInfo pluginInfo, PluginClass pluginClass, PluginProperties properties) {
+    this.pluginInfo = pluginInfo;
+    this.pluginClass = pluginClass;
+    this.properties = properties;
+  }
+
+  /**
+   * Returns the plugin information.
+   */
+  public PluginInfo getPluginInfo() {
+    return pluginInfo;
+  }
+
+  /**
+   * Returns the plugin class information.
+   */
+  public PluginClass getPluginClass() {
+    return pluginClass;
+  }
+
+  /**
+   * Returns the set of properties available for the plugin when the adapter was created.
+   */
+  public PluginProperties getProperties() {
+    return properties;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
This class is used inside `AdapterSpecification`, which we are planning to move it to api so that app can access it.